### DISCUSSION TITLE
weight norm used in tdnn module

### DISCUSSION THREAD

**Discussion by zhiyunfan** [pyannote/pyannote-audio#760] [Ideas]
Hello, bredin, I have tried to use pyannote.audio 1.1.1 to do speaker change detection. And I used the tdnn module to downsample my features. In the pyannote 1.0, tdnn is followed by a weight norm. When I tried to replace the weight norm with a batch norm, I got an obviously gain. And as far as I know, batch norm is most used in speaker related tasks. So, I suggest that maybe you can  try batch norm.

**Comment by hbredin** [comment]
Thanks @zhiyunfan for your feedback.

Did you train a model directly for speaker change detection ? or did you train a speaker embedding model and then used it for speaker change detection? 

Also note that upcoming pyannote.audio 2.0 (in develop branch) [uses batchnorm](https://github.com/pyannote/pyannote-audio/blob/995ddee906ab0befb633c8900469f42919a77bd0/pyannote/audio/models/embedding/xvector.py#L68-L75).
